sdks/python/apache_beam/transforms/window.py

```diff
@@ -245,10 +265,19 @@ def __init__(self, value, timestamp):
     self.value = value
     self.timestamp = Timestamp.of(timestamp)
 
-  def __cmp__(self, other):
+  def __eq__(self, other):
+    type_eq = type(self) == type(other)
+    value_eq = self.value == other.value
+    timestamp_eq = self.timestamp == other.timestamp
+    return type_eq and value_eq and timestamp_eq
```

Comment:
This will not work correctly, as the previous code relied on the short-circuiting behavior of "and".  Accessing `other.value` will not work if the type is not as we expected.